app/controllers/proctored_exams_controller.rb

```diff
@@ -32,20 +32,54 @@ def start_quiz
       proctor_code: params[:proctor_code],
     }.to_query
 
-    quiz = HTTParty.get(
-      "#{plugin.settings[:adhesion_url]}/api/proctored_exams?#{query}",
+    exam_request = HTTParty.get(
+      "#{plugin.settings[:adhesion_proctor_url]}/api/proctored_exams?#{query}",
       headers: headers,
       # verify: false,
-    )
-    if quiz.parsed_response["error"].present?
-      flash[:error] = quiz.parsed_response["error"]
+    ).parsed_response["exam_request"]
+
+    if exam_request.nil?
+      flash[:error] = "You do not have an exam that is ready to start."
+      redirect_to proctored_exams_path
+      return
+    end
+
+    matched_code = false
+    proctor_id = nil
+    proctor_name = nil
+    code = params[:proctor_code]
+    Account.find(exam_request["testing_center_id"]).users.find_each do |user|
+      cd = CustomData.find_by(user: user, namespace: "edu.au.exam")
+      if cd.nil? || cd.data["d"]["exam"]["proctor_code"] == code
+        matched_code = true
+        proctor_id = user.id
+        proctor_name = user.name
+      end
+    end
+
+    if !matched_code
+      flash[:error] = "Invalid proctor code."
       redirect_to proctored_exams_path
-    else
-      quiz = quiz.parsed_response
-      session[:is_proctored] = true
-      session[:proctor_access_code] = quiz["proctor_access_code"]
-      redirect_to course_quiz_take_path quiz["quiz"]["course_id"], quiz["quiz"]["exam_id"]
+      return
     end
+
+    query = {
+      proctor_id: proctor_id,
+      proctor_name: proctor_name,
+    }.to_query
+
+    # HTTParty is really bad and sends the put body wrong, so i send everything in the params
+    # RestClient does this better but I dont have RestClient so HTTParty it is.
+    HTTParty.put(
```

Comment:
Does the response need to be stored to a variable and something done with it? I see that you are doing a `.parsed_response` but I don't see the need to do that if you aren't needing to get the response.

Reply:
oops, I removed it.